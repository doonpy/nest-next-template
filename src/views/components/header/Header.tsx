import Head from 'next/head';
import React from 'react';
import { useSelector } from 'react-redux';

import PageReducer from '../../redux/reducers/PageReducer';

const Header: React.FC = () => {
  const pageReducer = PageReducer.getInstance();
  const pageState = useSelector<RootState, PageState>(
    (state) => state[pageReducer.getReducerKey()]
  );

  return (
    <Head>
      <title>{pageState.title}</title>
    </Head>
  );
};

export default Header;
