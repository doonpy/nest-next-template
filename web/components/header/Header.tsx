import Head from 'next/head';
import React from 'react';

interface ComponentProps {
  title: string;
}

const Header: React.FC<ComponentProps> = ({ title }) => {
  return (
    <Head>
      <title>{title}</title>
    </Head>
  );
};

export default Header;
