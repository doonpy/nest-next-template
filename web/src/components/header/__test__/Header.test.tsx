import { shallow } from 'enzyme';
import React from 'react';

import Header from '../Header';

describe('Header', () => {
  const title = 'test title';

  it('should render with correct title', () => {
    const component = shallow(<Header title={title} />);

    expect(component.contains(<title>{title}</title>)).toBeTruthy();
  });

  it('should render correctly', () => {
    const component = shallow(<Header title={title} />);

    expect(component).toMatchSnapshot();
  });
});
