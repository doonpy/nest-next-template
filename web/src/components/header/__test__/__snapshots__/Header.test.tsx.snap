// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<Head>
  <title>
    test title
  </title>
</Head>
`;
